rust/tw_aptos/src/transaction_builder.rs

```diff
@@ -0,0 +1,225 @@
+            self.sequence_number
+                .expect("sequence number must have been set"),
+            self.payload,
+            self.max_gas_amount,
+            self.gas_unit_price,
+            self.expiration_timestamp_secs,
+            self.chain_id,
+        )
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct TransactionFactory {
+    max_gas_amount: u64,
+    gas_unit_price: u64,
+    transaction_expiration_time: u64,
+    chain_id: u8,
+}
+
+impl TransactionFactory {
+    pub fn new(chain_id: u8) -> Self {
+        Self {
+            max_gas_amount: MAX_GAS_AMOUNT,
+            gas_unit_price: GAS_UNIT_PRICE,
+            transaction_expiration_time: 30,
+            chain_id,
+        }
+    }
+
+    pub fn new_from_protobuf(input: SigningInput) -> TransactionBuilder {
+        let factory = TransactionFactory::new(input.chain_id as u8)
+            .with_gas_unit_price(input.gas_unit_price)
+            .with_max_gas_amount(input.max_gas_amount)
+            .with_transaction_expiration_time(input.expiration_timestamp_secs);
+        match input.transaction_payload {
+            OneOftransaction_payload::transfer(transfer) => {
+                return factory.implicitly_create_user_account_and_transfer(AccountAddress::from_str(&transfer.to).unwrap(), transfer.amount);
+            }
+            OneOftransaction_payload::token_transfer(_) => {}
+            OneOftransaction_payload::create_account(_) => {}
+            OneOftransaction_payload::nft_message(_) => {}
+            OneOftransaction_payload::register_token(_) => {}
+            OneOftransaction_payload::liquid_staking_message(_) => {}
+            OneOftransaction_payload::token_transfer_coins(_) => {}
+            OneOftransaction_payload::None => {}
+        }
+        todo!()
+    }
+
+    pub fn with_max_gas_amount(mut self, max_gas_amount: u64) -> Self {
+        self.max_gas_amount = max_gas_amount;
+        self
+    }
+
+    pub fn with_gas_unit_price(mut self, gas_unit_price: u64) -> Self {
+        self.gas_unit_price = gas_unit_price;
+        self
+    }
+
+    pub fn with_transaction_expiration_time(mut self, transaction_expiration_time: u64) -> Self {
+        self.transaction_expiration_time = transaction_expiration_time;
+        self
+    }
+
+    pub fn with_chain_id(mut self, chain_id: u8) -> Self {
+        self.chain_id = chain_id;
+        self
+    }
+
+    pub fn get_max_gas_amount(&self) -> u64 {
+        self.max_gas_amount
+    }
+
+    pub fn get_gas_unit_price(&self) -> u64 {
+        self.gas_unit_price
+    }
+
+    pub fn get_transaction_expiration_time(&self) -> u64 {
+        self.transaction_expiration_time
+    }
+
+    pub fn payload(&self, payload: TransactionPayload) -> TransactionBuilder {
+        self.transaction_builder(payload)
+    }
+
+    pub fn entry_function(&self, func: EntryFunction) -> TransactionBuilder {
+        self.payload(TransactionPayload::EntryFunction(func))
+    }
```

Comment:
So for example, this method could return `RawTransaction`.
Or do you have a plan to add some logic into `TransactionBuilder`? Then I see why you used `TransactionBuilder` here